### ISSUE TITLE
[Java] A false negative about the rule SingularField

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3680]
**Affects PMD Version:6.41.0** 

**Rule:SingularField**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#singularfield>

**Description:**
Hi, I found a false negative about the rule SingularField. Please refer to the following case, PMD should have reported a false negative at line 2. 

**Code Sample demonstrating the issue:**
```java
class Impl {
    private int var;  // should report a warning here
    public int foo(int y) {
        if(true) {
            var = y + 5;
        } else {
            var = y + 5;
        }
        return var;
    }
}
```

**Expected outcome:**

PMD should report a violation at line 2, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

An equivalent case can be deteced is below:

```java
private int var;  // can report a warning here
public int foo(int y) {
    for(; true; ) {
        var = y + 5;
        break;
    }
    return var;
}
```
